python-package/lightgbm/basic.py

```diff
@@ -1840,7 +1883,17 @@ def set_init_score(self, init_score):
         """
         self.init_score = init_score
         if self.handle is not None and init_score is not None:
-            init_score = list_to_1d_numpy(init_score, np.float64, name='init_score')
+            if is_1d_collection(init_score):
+                init_score = list_to_1d_numpy(init_score, np.float64, name='init_score')
+            elif is_2d_collection(init_score):
+                init_score = data_to_2d_numpy(init_score, np.float64, name='init_score')
+                init_score = init_score.ravel()
+                self.init_score = init_score
+            else:
+                raise TypeError(
+                    'init_score must be list, numpy 1-D array or pandas Series.\n'
+                    'In multiclass classification init_score must be list, numpy 2-D array or pandas DataFrame.'
+                )
             self.set_field('init_score', init_score)
```

Comment:
What about `set_field()` method? Should it also support 2-D collections for multiclass task?

Reply:
Seems like the dataset fields are vectors, or do you mean making the collection 1-D inside that method?

Reply:
Yeah, cpp side accents only flatten vectors even for multiclass task. I mean, should `set_field()` method accept 2D matrix as init score for multiclass task? Actually, it means just move all (or some) your new logic from `set_init_score` to `if field_name == 'init_score':` branch of  `set_field()` method. So that users will be able pass 2D matrices to both methods.

Reply:
I don't have a preference, if you think it makes more sense for this to be inside `set_field` I can move it there.

Reply:
Why users are able to pass 2-D object to `set_init_score()` and not able to do so in `set_field('init_score', ...)`? I believe it makes sense to unify all interfaces.

Reply:
Done in ad4495942d486435a7bfa2a5e382dc661c2e6f7a